dbuf/dbuf/client.py

```diff
@@ -0,0 +1,171 @@
+import random
+import asyncio
+import aiohttp
+import collections
+import struct
+
+import hailtop.utils as utils
+from hailtop.config import get_deploy_config
+
+
+class DBufClient:
+    async def __aenter__(self):
+        self.aiosession = aiohttp.ClientSession(raise_for_status=True,
+                                                timeout=aiohttp.ClientTimeout(total=60))
+        return self
+
+    async def __aexit__(self, exc_type, exc, tb):
+        await self.aiosession.close()
+
+    def __init__(self,
+                 leader_name,
+                 id=None,
+                 max_bufsize=10*1024*1024 - 1,
+                 deploy_config=None,
+                 rng=None):
+        if not deploy_config:
+            deploy_config = get_deploy_config()
+        if rng is None:
+            rng = random.Random()
+        self.rng = rng
+        self.deploy_config = deploy_config
+        self.root_url = deploy_config.base_url(leader_name)
+        self.session_url = None if id is None else f'{self.root_url}/s/{id}'
+        self.aiosession = None
+        self.id = id
+        self.offs = []
+        self.sizes = []
+        self.cursor = 0
+        self.max_bufsize = max_bufsize
+
+    async def create(self):
+        resp = await utils.request_retry_transient_errors(
+            self.aiosession,
+            'POST',
+            f'{self.root_url}/s')
+        assert resp.status == 200
+        self.id = int(await resp.text())
```

Comment:
Prefer this to be json.

Reply:
This seems like more code for the same effect and unnecessarily invokes a json parser?
```python
id = await resp.json()
assert isinstance(id, int)
self.id = id
```